addons/mail/static/src/js/composers/basic_composer.js

```diff
@@ -416,6 +416,7 @@ var BasicComposer = Widget.extend({
             attachment_ids: _.pluck(this.get('attachment_ids'), 'id'),
             partner_ids: _.uniq(_.pluck(this._mentionManager.getListenerSelection('@'), 'id')),
             canned_response_ids: _.uniq(_.pluck(this._mentionManager.getListenerSelections()[':'], 'id')),
+            channel_ids: _.uniq(_.pluck(this._mentionManager.getListenerSelection('#'), 'id')),
```

Comment:
update `getListenerSelection` so that it escapes the composer input val:
```js
...
var escapedVal = _.escape(thiis._composer.$input.val());
var inputMentions = escapedVal.match(new RegExp(delimiter+'[^ ]+(?= |&nbsp;)', 'g'));
...
```

Reply:
this should fix an issue in 12.0, in which partners with special characters do not receive needaction when mentioned